docs/encyclopedia/application-message-passing.mdx

```diff
@@ -26,57 +28,85 @@ keywords:
 - temporal sdk message passing
 ---
 
+import PrettyImage from '@site/src/components/pretty-image/PrettyImage';
 import { RelatedReadContainer, RelatedReadItem } from '@site/src/components/related-read/RelatedRead';
 
-Temporal applications can receive and provide data through messages.
+Workflows can be thought of as stateful web services that can receive messages. The Workflow can have powerful message handlers akin to endpoints that can react to the incoming messages in combination with the current state of the Workflow.
 Temporal supports three types of messages: Signals, Queries, and Updates.
+To define these, consider these messages from the perspective of the client making the request:
 
-## Signals {#signals}
+- Queries are read requests. They can read the current state of the Workflow but cannot block in doing so.
+- Signals are write requests. They cause changes in the running Workflow, but you cannot await any response or error.
+- Updates are interactive write requests, wherein the sender of the Update tracks completion and waits for a response or an error.
 
-**What is a Signal in the context of Temporal?**
+## How to choose between Signals, Updates, and Queries? {#choosing-messages}
 
-A Signal is an asynchronous request to a [Workflow Execution](/workflows#workflow-execution).
+### For write requests
 
-<RelatedReadContainer>
-    <RelatedReadItem path="/cli/workflow#signal" text="Send a Signal using the Temporal CLI" archetype="feature-guide" />
-    <RelatedReadItem path="/develop/go/message-passing#signals" text="Send and handle Signals with the Go SDK" archetype="feature-guide" />
-    <RelatedReadItem path="/develop/java/message-passing" text="Send and handle Signals with the Java SDK" archetype="feature-guide" />
-    <RelatedReadItem path="/develop/php/message-passing#signals" text="Send and handle Signals with the PHP SDK" archetype="feature-guide" />
-    <RelatedReadItem path="/develop/python/message-passing#signals" text="Send and handle Signals with the Python SDK" archetype="feature-guide" />
-    <RelatedReadItem path="/develop/typescript/message-passing#signals" text="Send and handle Signals with the TypeScript SDK" archetype="feature-guide" />
-    <RelatedReadItem path="/develop/dotnet/message-passing#define-a-signal" text="Send and handle Signals with the .NET SDK" archetype="feature-guide" />
-</RelatedReadContainer>
+From the above descriptions, it may sound like Updates can do all the things that Queries and Signals can do. But, unlike Signals, Updates must be synchronous, that is, they must wait for the worker running the Workflow to acknowledge the request.
```

Comment:
```suggestion
From the above descriptions, it may sound like Updates can do all the things that Queries and Signals can do. But, unlike Signals, Updates must wait for the worker running the Workflow to acknowledge the request.
```

I'm not sure we should define "synchronous" here, since that word (and "asynchronous") have multiple similar meanings that are relevant to Temporal. After all, Update can also be viewed as "asynchronous" in the sense that an acknowledgement response can be received before the result is available. Indeed, there is some precedent for referring to that latter style as "async update" (although I propose that we also do not allow that terminology to get into our public-facing docs).

Reply:
I think we need a short term for this, and  synchronous really is a term of art for this in software engineering ("Interactive" was my half-attempt to be more nuanced, but maybe it doesn't land) especially in the sense that the remote worker must be present.  And when we add durable-on-admitted, we can call it "async updates" as shorthand.  Going to keep it in the absence of an alternative term.
My theory is that the waiting-for-just-validation is niche enough that it's okay to gloss over that slightly at this level.